File: handin1/me_readers.js
// 1 READ CSV FILE
const fs = require("fs");

fs.readFile("me.csv", "utf8", (err, data) => {
  if (err) throw err;

  const lines = data.split("\n");
  const header = lines[0].split(",");
  const values = lines[1].split(",");

  const result1 = {};
  header.forEach((item, index) => {
    result1[item] = values[index];
  });
  console.log("Reading CSV file...");
  console.log(JSON.stringify(result1));
});

// 2 READ JSON FILE
fs.readFile("me.json", "utf8", (err, data) => {
  if (err) throw err;

  const result2 = JSON.parse(data);

  console.log(`Reading JSON file...\n${JSON.stringify(result2)}`);
});

// 3 READ TXT FILE
fs.readFile("me.txt", "utf8", (err, data) => {
  if (err) throw err;

  const lines = data.split("\n");
  const result3 = {};

  for (const line of lines) {
    const [key, value] = line.split(": ");
    result3[key] = value;
  }

  console.log(`Reading TXT file...\n${JSON.stringify(result3)}`);
});

// 4 READ XML FILE
const xml2js = require("xml2js");

fs.readFile("me.xml", "utf8", (err, data) => {
  if (err) throw err;

  xml2js.parseString(data, (err, result4) => {
    if (err) throw err;

    console.log(`Reading XML file...\n${JSON.stringify(result4)}`);
  });
});

// 5 READ YAML FILE
const yaml = require("js-yaml");

fs.readFile("me.yaml", "utf8", (err, data) => {
  if (err) throw err;

  try {
    const doc = yaml.load(data);
    console.log(`Reading YAML file...\n${JSON.stringify(doc)}`);
  } catch (e) {
    console.error(e);
  }
});

File: handin1/me_readers.py
# 1 READ CSV FILE

#import yaml
import xml.etree.ElementTree as ET
import json
import csv
print("Reading CSV file...\n")

with open("me.csv") as file:
    reader = csv.reader(file)
    headers = next(reader)

    for row in reader:
        name, age, *hobbies = row
        print("Name:", name)
        print("Age:", age)
        print("Hobbies:", hobbies)
        print("\n\n")


# 2 READ JSON FILE

print("Reading JSON file...\n")

with open("me.json") as file:
    data = json.load(file)

person = data["person"]
name = person["name"]
age = person["age"]
hobbies = person["hobbies"]

print("Name:", name)
print("Age:", age)
print("Hobbies:", hobbies)
print("\n\n")


# 3 READ TXT FILE

print("Reading TXT file...\n")
with open("me.txt") as file:
    lines = file.readlines()

data = {}
for line in lines:
    key, value = line.strip().split(": ")
    data[key] = value

name = data["Name"]
age = int(data["Age"])
hobbies = data["Hobbies"].split(", ")

print("Name:", name)
print("Age:", age)
print("Hobbies:", hobbies)
print("\n\n")


# 4 READ XML FILE

print("Reading XML file...\n")

tree = ET.parse("me.xml")
root = tree.getroot()

name = root.find("name").text
age = int(root.find("age").text)
hobbies = [hobby.text for hobby in root.find("hobbies").findall("hobby")]

print("Name:", name)
print("Age:", age)
print("Hobbies:", hobbies)
print("\n\n")


# 5 READ YAML FILE

""" print("Reading YAML file...\n")

with open("me.yaml") as file:
    data = yaml.safe_load(file)

name = data["person"]["name"]
age = data["person"]["age"]
hobbies = data["person"]["hobbies"]

print("Name:", name)
print("Age:", age)
print("Hobbies:", hobbies) """
